### DISCUSSION TITLE
Images Loading Halfway on IPad

### DISCUSSION THREAD

**Discussion by st11x** [dimsemenov/PhotoSwipe#1941] [Q&A]
Hi,

I have this issue on the IPad where an image will not load completely. This is typically on images that are dynamically added to the DOM before a thumbnail is clicked to bring up the gallery. The image that you clicked on will load halfway, but the left and right (those preloaded) ones will be okay.

The image in question will be half blurred because the library shows the thumbnail scaled up first and it replaces it eventually. Then later, the part where it was blurred gets removed completely; leaving the image cut off.

I tried taking a screenshot on the iPad to paste into this message, the paste didn’t work but upon returning to the gallery, the image is loaded in full. 

It sounds like a webkit thing, was wondering if anyone has run into this problem and how did you solve that? 

Thanks
Matt

**Comment by dimsemenov** [comment]
Can you reproduce this issue on the main photoswipe demo too? Very hard to tell what might be causing this, are you using the latest iOS?

**Reply by st11x** [reply]
I couldn’t reproduce it on the main demo. I also couldn’t reproduce it if the images before there the init was called.
I experimented with destroying the instance and calling init again but that doesn’t help.

I’m on the latest IOS 15.x. 

One other (maybe unrelated) observation is that Photoswipe doesn’t work on a browser like Via, even the demo ones on the main page. 

thanks
  Matt

**Comment by dimsemenov** [comment]
Possibly images to large? You may try `will-change` style on `img.pswp__img` element.

**Reply by st11x** [reply]
My images are JPGs with a max longest side at 1536, it is about half of the 3000px listed on the site, so I don't think that's the problem.

When the page is hosted locally, I don't seem to have that problem. I'll try your suggestions. 

I got some screenshots. The first shows the first time I try to open the image. It goes half way and stops, the rest is just a blur.

![Blur](https://user-images.githubusercontent.com/7398499/175830082-137d2294-b41f-4478-9d72-f4761f94dc65.jpg)

If I close it and reopen the same image, it is now cut off


![Blur2](https://user-images.githubusercontent.com/7398499/175830118-984fbf61-de6c-42e9-9436-0ae571ef923f.jpg)


Thanks
  Matt
